packages/jsii-pacmak/lib/targets/python.ts

```diff
@@ -45,8 +47,13 @@ export default class Python extends Target {
   public async generateCode(outDir: string, tarball: string): Promise<void> {
     await super.generateCode(outDir, tarball);
 
+    // Using a static variable as a lock to prevent racing. Since blackPath() uses
+    // Promise APIs from fs and os modules (that use libuv), an additional lock is required.
+    if (Python.BLACK_PATH === undefined) {
+      Python.BLACK_PATH = this.blackPath();
+    }
```

Comment:
I'd have wrapped this caching within `this.blackPath()` instead, possibly even making `blackPath()` static while there. This would "guard" against incorrectly calling `blackPath` multiple times?

Reply:
I like the layering that I have already. It lets me test `blackPath()`. I'll have to possibly mock a few extra things to get the correct test behaviour of current `blackPath()` (see unit tests that are already there).

I don't expect this to be called again from anywhere else. The method organization is simply readability and testability.

How strongly do you feel?

Reply:
Not very :) And you've convinced me there was enough reason to leave it like so.

Reply:
😊